src/WarcraftLegacies.Source/Quests/Dalaran/QuestTheramore.cs

```diff
@@ -62,6 +63,14 @@ protected override void OnComplete(Faction completingFaction)
         unit.SetOwner(Player(PLAYER_NEUTRAL_AGGRESSIVE));
     }
 
+    protected override void OnFail(Faction completingFaction)
```

Comment:
You have an `OnFail` method so you need a `PenaltyDescription` and a `PenaltyFlavour`, similar to the reward versions.